src/test/java/org/scijava/io/location/resolve/FileLocationResolverTest.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * #%L
+ * SciJava Common shared library for SciJava software.
+ * %%
+ * Copyright (C) 2009 - 2017 Board of Regents of the University of
+ * Wisconsin-Madison, Broad Institute of MIT and Harvard, Max Planck
+ * Institute of Molecular Cell Biology and Genetics, University of
+ * Konstanz, and KNIME GmbH.
+ * %%
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ * 
+ * 1. Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ * 
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDERS OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ * #L%
+ */
+
+package org.scijava.io.location.resolve;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+
+import org.junit.Test;
+import org.scijava.Context;
+import org.scijava.io.location.FileLocation;
+import org.scijava.io.location.Location;
+
+/**
+ * Test for {@link FileLocationResolver}.
+ * 
+ * @author Gabriel Einsdorf
+ */
+public class FileLocationResolverTest {
+
+	Context ctx = new Context(LocationService.class);
+	LocationService resolver = ctx.getService(
+		LocationService.class);
+
+	@Test
+	public void testStringResolve() throws URISyntaxException {
+		String uri = getClass().getResource("/").toString();
+		Location loc = resolver.resolve(uri);
+		assertTrue(loc instanceof FileLocation);
+		assertEquals(uri, loc.getURI().toString());
+	}
+
+	@Test
+	public void testURIResolve() throws URISyntaxException {
+		URI uri = new URI(getClass().getResource("/").toString());
+		Location loc = resolver.resolve(uri);
+		assertTrue(loc instanceof FileLocation);
```

Comment:
Does that really work always? It seems dicey. Maybe use `ClassUtils.getLocation` instead? Or just hardcode something like `new File(".").getAbsolutePath()` which is guaranteed to work on every platform?